core/experiments/assets/js/admin/behaviors/experiments-messages.js

```diff
@@ -168,4 +170,60 @@ export default class ExperimentsDependency {
 
 		return clone.join( glue ) + finalGlue + lastItem;
 	}
+
+	setDialog( experiment, dialog ) {
+		return elementorCommon.dialogsManager.createWidget( 'confirm', {
+			id: 'e-experiments-messages-dialog',
+			headerMessage: dialog.headerMessage,
+			message: dialog.message,
+			position: {
+				my: 'center center',
+				at: 'center center',
+			},
+			strings: {
+				confirm: dialog.confirm,
+				cancel: dialog.cancel,
+			},
+			hide: {
+				onOutsideClick: false,
+				onBackgroundClick: false,
+				onEscKeyPress: false,
+			},
+			onConfirm: () => {
+				dialog.onConfirm();
+			},
+			onCancel: () => {
+				this.setExperimentState( experiment.name, dialog.initialState )
+			},
+		} );
+	}
+
+	setDialogData( experiment, experimentId, state ) {
+		return state === STATE_INACTIVE
```

Comment:
use regular if else, and then create build the messages inside it:

```JS
if ( active ) {
	// Translators: %1$s: Experiment title, %2$s: Message content
	const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
			.replace( '%1$s', `<strong>${ title }</strong>` )
			.replace( '%2$s', `<strong>${ message }</strong>` );

	return {
		message,
		...
	}
} else {
	const experimentsList = this.joinDepenednciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ));

	const message = __( 'While deactivating %1$s, %2$s.', 'elementor' )
			.replace( '%1$s', `<strong>${ title }</strong>` )
			.replace( '%2$s', `<strong>${ message }</strong>` );
	return {
		 ...
	}
}
```

Reply:
maybe not relevant

Reply:
Not relevant